From: Zhong Jianxin <azuwis@gmail.com>
Date: Wed, 11 Nov 2015 00:22:31 +0800
Subject: Prioritize last pieces to 16M

---
 src/core/qtlibtorrent/qtorrenthandle.cpp | 8 +++++++-
 1 file changed, 7 insertions(+), 1 deletion(-)

diff --git a/src/core/qtlibtorrent/qtorrenthandle.cpp b/src/core/qtlibtorrent/qtorrenthandle.cpp
index a1cf5a3..1ca9563 100644
--- a/src/core/qtlibtorrent/qtorrenthandle.cpp
+++ b/src/core/qtlibtorrent/qtorrenthandle.cpp
@@ -726,7 +726,13 @@ void QTorrentHandle::prioritize_first_last_piece(int file_index, bool b) const
 
     QPair<int, int> extremities = get_file_extremity_pieces(*tf, file_index);
     piece_priority(extremities.first, prio);
-    piece_priority(extremities.second, prio);
+    int end_size = 16 * 0x100000;
+    int piece_num = extremities.second;
+    while (end_size > 0 && piece_num > 1) {
+        piece_priority(piece_num, prio);
+        end_size -= piece_length();
+        piece_num--;
+    }
 }
 
 void QTorrentHandle::prioritize_first_last_piece(bool b) const
